frontend/web/components/release-pipelines/StageSummaryData.tsx

```diff
@@ -0,0 +1,96 @@
+import moment from 'moment'
+
+import {
+  Features,
+  StageAction,
+  StageTrigger,
+  StageTriggerType,
+} from 'common/types/responses'
+
+import Icon from 'components/Icon'
+
+import { renderActionDetail } from './FlagActionDetail'
+
+export interface StageSummaryDataProps {
+  stageActions?: StageAction[]
+  stageTrigger?: StageTrigger
+  featureInStage?: Features[number]
+}
+
+const StageSummaryData = ({
+  featureInStage,
+  stageActions,
+  stageTrigger,
+}: StageSummaryDataProps) => {
+  const isWaitType = stageTrigger?.trigger_type === StageTriggerType.WAIT_FOR
+
+  const waitForTimeoutDuration = moment.duration(
+    stageTrigger?.trigger_body?.wait_for,
+  )
+
+  const timeDifference = featureInStage?.created_at
+    ? moment().diff(moment(featureInStage.created_at))
+    : 0
+
+  const timeoutDuration = stageTrigger?.trigger_body?.wait_for
+    ? waitForTimeoutDuration.asMilliseconds()
+    : 0
+
+  const timeRemaining = timeoutDuration - timeDifference
+  const timeRemainingDuration = moment.duration(Math.max(0, timeRemaining))
+
+  const isTimeLeft = timeRemainingDuration.asMilliseconds() > 0
+  const isActionPending = isWaitType && isTimeLeft
```

Comment:
<img width="1326" height="469" alt="image" src="https://github.com/user-attachments/assets/4841ef0a-e64c-43b3-8f63-9c16954172f4" />

I have this on preview (my local is not correctly wired it seems). Could you re-check this part please `moment(featureInStage.created_at)` and see if maybe we need to add the format.

It seems to also impact the checking of the action

Reply:
What do you mean by format?